crates/biome_js_analyze/src/lint/nursery/no_document_cookie.rs

```diff
@@ -0,0 +1,111 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_js_syntax::{
+    global_identifier, AnyJsAssignment, AnyJsAssignmentPattern, JsAssignmentExpression,
+};
+use biome_rowan::AstNode;
+
+use crate::services::semantic::Semantic;
+
+declare_lint_rule! {
+    /// Disallow use `document.cookie` directly.
+    ///
+    /// It's not recommended to use document.cookie directly as it's easy to get the string wrong.
+    /// Instead, you should use the [Cookie Store API](https://developer.mozilla.org/en-US/docs/Web/API/CookieStore) or a [cookie library](https://www.npmjs.com/search?q=cookie).
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// document.cookie = "foo=bar";
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// document.cookie += "; foo=bar";
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// await cookieStore
+    ///   .set({
+    ///     name: "foo",
+    ///     value: "bar",
+    ///     expires: Date.now() + 24 * 60 * 60,
+    ///     domain: "example.com",
+    /// })
+    /// ```
+    ///
+    /// ```js
+    /// const array = document.cookie.split("; ");
+    /// ```
+    ///
+    /// ```js
+    /// import Cookies from 'js-cookie';
+    ///
+    /// Cookies.set('foo', 'bar');
+    /// ```
+    ///
+    pub NoDocumentCookie {
+        version: "next",
+        name: "noDocumentCookie",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintUnicorn("no-document-cookie")],
+    }
+}
+
+impl Rule for NoDocumentCookie {
+    type Query = Semantic<JsAssignmentExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let left = node.left().ok()?;
+
+        let static_assignment = match &left {
+            AnyJsAssignmentPattern::AnyJsAssignment(AnyJsAssignment::JsStaticMemberAssignment(
+                static_assignment,
+            )) => static_assignment,
+            _ => {
+                return None;
+            }
+        };
```

Comment:
The `left` function should have methods like `as_js_assignment_pattern` and then `as_js_static_member_assignment`. Use these methods instead of pattern matching.

```rs
let static_assignment = left.as_js_assignment_pattern()?.as_js_static_member_assignment()?;
```

Reply:
This section has been rewritten to support bracket notation such as `document[“cookie”]' at https://github.com/biomejs/biome/pull/4204/commits/a0ec05fff78ed2337eca5a0ad8f9207d7208432b.

It currently looks like this.

https://github.com/biomejs/biome/blob/a0ec05fff78ed2337eca5a0ad8f9207d7208432b/crates/biome_js_analyze/src/lint/nursery/no_document_cookie.rs#L115-L123

https://github.com/biomejs/biome/blob/a0ec05fff78ed2337eca5a0ad8f9207d7208432b/crates/biome_js_analyze/src/lint/nursery/no_document_cookie.rs#L86-L91